extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -841,6 +865,52 @@ export class AmpStoryPage extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Builds and appends a message and icon to play the story on tap.
+   * This message is built when the playback failed (data saver, low battery
+   * modes, ...).
+   * @private
+   */
+  buildAndAppendPlayMessage_() {
+    const localizationService = Services.localizationService(this.win);
+
+    this.playMessageEl_ = buildPlayMessageElement(this.element);
+    const labelEl =
+        this.playMessageEl_.querySelector('.i-amphtml-story-page-play-label');
+    labelEl.textContent = localizationService.getLocalizedString(
+        LocalizedStringId.AMP_STORY_PAGE_PLAY_VIDEO);
+
+    this.playMessageEl_.addEventListener('click', () => {
+      this.togglePlayMessage_(false);
+      this.mediaPoolPromise_
+          .then(mediaPool => mediaPool.blessAll())
```

Comment:
I can show you tomorrow on the iPhone 7 or 5 :)
I think it's a product question, because we might need to bless the video for the `auto-advance` case. In the demo I posted, if we remove the bless and wait for the story to auto advance, we'd have to tap the button on the second page to play the video.
However, if we bless it, it's going to play if we tap OR wait. I opted for this since it provides a more consistent experience. It's fine on the iPhone 7, but if you think it's too much of a performance hit, we can remove it.

We could play the videos and then bless everything, but I'm afraid the bless actions would freeze the playing video. :((

Reply:
+1.  It's better to bless if possible, I just want to make sure the performance is not a blocker.